src/lib/core/ripple/ripple.ts

```diff
@@ -86,36 +77,71 @@ export class MatRipple implements OnChanges, OnDestroy {
+    if (!this.disabled && this._isInitialized) {
+      this._rippleRenderer.setTriggerElement(trigger);
+    }
+  }
+  private _trigger: HTMLElement | null;
 
-  /** Whether foreground ripples should be visible outside the component's bounds. */
-  @Input('matRippleUnbounded') unbounded: boolean;
+  /**
+   * Whether click events will not trigger the ripple. Ripples can be still launched manually
+   * by using the `launch()` method.
+   */
+  @Input('matRippleDisabled')
+  get disabled() { return this._disabled || !!this._globalOptions.disabled; }
+  set disabled(value: boolean) {
+    this._disabled = value;
+
+    // The events for the trigger element haven't been added if ripples were disabled. This means
+    // that the events for the trigger element need to be registered if ripples are enabled again.
+    if (!this.disabled && this._isInitialized) {
+      this._rippleRenderer.setTriggerElement(this._trigger);
+    }
+  }
+  private _disabled: boolean = false;
 
   /** Renderer for the ripple DOM manipulations. */
   private _rippleRenderer: RippleRenderer;
 
   /** Options that are set globally for all ripples. */
   private _globalOptions: RippleGlobalOptions;
 
-  constructor(
-    elementRef: ElementRef,
-    ngZone: NgZone,
-    platform: Platform,
-    @Optional() @Inject(MAT_RIPPLE_GLOBAL_OPTIONS) globalOptions: RippleGlobalOptions
-  ) {
-    this._rippleRenderer = new RippleRenderer(elementRef, ngZone, platform);
-    this._globalOptions = globalOptions ? globalOptions : {};
+  /** Reference to the directive's host HTMLElement. */
+  private _hostElement: HTMLElement;
+
+  /** Whether ripple directive is initialized and the input bindings are set. */
+  private _isInitialized: boolean = false;
 
-    this._updateRippleRenderer();
+  constructor(elementRef: ElementRef,
+              ngZone: NgZone,
+              platform: Platform,
+              @Optional() @Inject(MAT_RIPPLE_GLOBAL_OPTIONS) globalOptions: RippleGlobalOptions) {
+
+    this._globalOptions = globalOptions || {};
+    this._hostElement = elementRef.nativeElement;
+    this._rippleRenderer = new RippleRenderer(this, ngZone, elementRef, platform);
   }
 
-  ngOnChanges(changes: SimpleChanges) {
-    if (changes['trigger'] && this.trigger) {
-      this._rippleRenderer.setTriggerElement(this.trigger);
-    }
+  ngOnInit() {
+    this._isInitialized = true;
 
-    this._updateRippleRenderer();
+    // The trigger element might have been set before the `ngOnInit` lifecycle hook, but the
+    // events haven't been registered yet. This has been deferred to the `ngOnInit` lifecycle hook,
+    // because the events should be only registered if the ripples aren't disabled initially.
+    // Also if there is no trigger element, that has been specified explicitly through the input,
+    // the default trigger element will be the directive's host element.
+    this.trigger = this._trigger === undefined ? this._hostElement : this.trigger;
```

Comment:
Doing this just so the getter runs is somewhat side-effect-ey. Why not have a private method that does the attachment both here and insider the setter?